src/sage/manifolds/calculus_method.py

```diff
@@ -275,7 +294,9 @@ def simplify(self, expression, method=None):
             method = self._current
         return self._simplify_dict[method](expression)
 
-    def is_trivial_zero(self, expression, method=None):
+    def is_trivial_zero(
+        self, expression: Expression, method: Optional[CalculusMethodName] = None
+    ) -> bool:
```

Comment:
Also here, the type declaration `Expression` is wrong. It depends on `method` what type is allowed.